internal/agent/validate_test.go

```diff
@@ -24,6 +24,21 @@ var _ = Describe("Validate", func() {
 	Context("Validate", func() {
 		var yaml string
 
+		Context("With a really long config string", func() {
+			BeforeEach(func() {
+				yaml = `#cloud-config
+users:
+  - name: kairos
+    passwd: kairos
+vpn:
+  network_token: "dssdnfjkldashfkjhasdkhfkasjdhfkjhasdjkfhaksjdhfkjashjdkfhioreqwhfuihqweruifhuewrbfhuewrfuyequfhuiehuifheqrihfuiqrehfuirqheiufhreqiuhfuiqheiufhqeuihfuiqrehfiuhqreuifrhiuqehfiuhqeirhfiuewhrfhqwehfriuewhfuihewiuhfruewhrifhwiuehrfiuhweiurfhwueihrfuiwehufhweuihrfuiwerhfuihewruifhewuihfiouwehrfiouhwei"
+`
+			})
+			It("validates", func() {
+				Expect(Validate(yaml)).ToNot(HaveOccurred())
```

Comment:
Maybe we should also check that it was treated as the actual sources? Just in case a different path was followed that also didn't produce an error (but did the wrong thing).